### PR TITLE
Fix #19: Add ClientBuilder to create Client

### LINKED ISSUE #19: Brainstorm ways to make Client constructor easier to use
**Motivation:** Currently Client constructor at least expects a collection name and a verification logic or None but we want to brainstorm other ways where user does not have to provide None in the Client to signify using Default Verification

### PR DESCRIPTION
**Pull request by Vishwa-Mozilla**
Add ClientBuilder to ease Client creation and very customizable while making the code very concise and clear

### CODE REVIEW

**Comment by Vishwa-Mozilla** [comment]
This code change is based off https://github.com/Vishwa-Mozilla/remote-settings-client/pull/41

File: rs-client-demo/src/main.rs [hunk 1]
```diff
@@ -27,133 +27,133 @@ fn main() {
 
     println!("Fetching records using RS client with default Verifier");
 
-    let client = Client::create_with_collection("url-classifier-skip-urls", None);
+    let client = ClientBuilder::new().collection_name("url-classifier-skip-urls").build();
```
Comment (leplatrem) [inline rs-client-demo/src/main.rs:30]: I kinda like `Client::builder().collection_name("...").build()`

File: rs-client-demo/src/main.rs [hunk 2]
```diff
@@ -27,133 +27,133 @@ fn main() {
...
 
-    match client.get(expected) {
+    match client.get() {
```
Comment (leplatrem) [inline rs-client-demo/src/main.rs:32]: Isn't this change part of #26 ? (I'm reviewing a single commit here)

File: src/client.rs [hunk 3]
```diff
@@ -59,21 +59,72 @@ pub struct Collection {
...
+    bucket_name: String,
+    collection_name: String,
+    verifier: Box<dyn Verification>,
+}
+
+impl ClientBuilder {
+
+    /// Constructs a new `ClientBuilder`.
+    ///
+    /// This is the same as `Client::builder()`.
```
Comment (leplatrem) [inline src/client.rs]: I don't see `Client::builder()` defined

File: src/client.rs [hunk 4]
```diff
@@ -59,21 +59,72 @@ pub struct Collection {
...
 /// Handles requests to Remote-Settings
 /// # Examples
 /// Create Client with collection_name and without custom Verifier
 /// ```rust
 /// # use remote_settings_client::{SignatureError, Verification};
-/// # use remote_settings_client::{Client, Collection};
+/// # use remote_settings_client::{Client, ClientBuilder, Collection};
 /// # fn main() {
-///   let client = Client::create_with_collection("collection_name", None);
+///   let client = ClientBuilder::new().collection_name("collection_name").build();
```
Comment (leplatrem) [inline src/client.rs:123]: using `Client::builder()` would save us imports

File: src/client.rs (hunk 3 above)
Comment (Vishwa-Mozilla) [reply src/client.rs]: done

File: rs-client-demo/src/main.rs (hunk 1 above)
Comment (Vishwa-Mozilla) [reply rs-client-demo/src/main.rs:30]: yes, I have made the change

**Comment by Vishwa-Mozilla** [comment]
I had to force-push this time (I accidentally introduced some unnecessary commits when rebasing my changes against master branch)

File: src/client.rs [hunk 5]
```diff
@@ -303,14 +312,13 @@ mod tests {
...
-            Client::create_with_server_collection(
-                &mock_server_address,
-                "url-classifier-skip-urls",
-                Some(Box::new(VerifierWithVerificatonError {})),
-            ),
+            Client::builder()
+            .server_url(&mock_server_address)
+            .collection_name("url-classifier-skip-urls")
+            .verifier(Box::new(VerifierWithVerificatonError {}))
+            .build(),
```
Comment (leplatrem) [inline src/client.rs:321]: Any idea why this one is not indented like the others?

Comment (Vishwa-Mozilla) [reply src/client.rs]: oops, fixed in latest commit

**Pull request merged**